### DISCUSSION TITLE
Why graphcache no caching in custom implementation?

### DISCUSSION THREAD

**Discussion by Dindaleon** [urql-graphql/urql#2931] [Q&A]
I am using Qwik for my frontend instead of react. I implemented graphcache as indicated in urql's documentation:

```
import { createClient, dedupExchange, fetchExchange } from 'urql';
import { cacheExchange } from '@urql/exchange-graphcache';

const client = createClient({
  url: 'http://localhost:3000/graphql',
  exchanges: [dedupExchange, cacheExchange({}), fetchExchange],
});
```

However, nothing is being cached. I checked it with urql dev tools for chrome and all I get are `cacheoutcome = miss`

I have googled for examples but all of them use React. Are there any in depth tutorials for using urql's graphcache when not using React/Svelte/Vue?

**Comment by przemek-git** [comment]
Hey! Could you also show how you're doing queries/mutations which results are supposed to be cached? 

You seem to be using `urql` package which is meant to be used with React, for any unsupported framework I would use [Core / Node.js](https://formidable.com/open-source/urql/docs/basics/core/), so import directly from `@urql/core` (even though all the framework packages reexport things from core package, I would still try with direct core imports). 

One more thing - are you getting `acheoutcome = miss` in "Explorer" tab of urql dev tools or rather you're trying to simply `console.log` cache in browser dev tools? I cannot see `devtoolsExchange` in your client exchanges, which is needed for urql dev tools.

**Reply by Dindaleon** [reply]
I am using `@urql/core`

I am using it like follows:

```
import {
  QRL,
  ResourceReturn,
  useContext,
  useResource$,
  useStore,
} from '@builder.io/qwik'
import { isServer } from '@builder.io/qwik/build'
import { AuthStateContext } from '~/context'
import { AuthStateProps } from '~/types'
import { graphqlClient } from '../client'

export const useQuery = (q: QRL<() => any>, vars: any): ResourceReturn<any> => {
  const auth: AuthStateProps = useContext(AuthStateContext)

  const queryStore = useStore({
    recursive: 'true',
    data: false,
  })

  return useResource$<any>(async ({ track, cleanup }) => {
    const qq = await q()

    const [client] = await Promise.all([
      graphqlClient.new({
        isServer: isServer,
        auth: {
          accessToken: auth.accessToken,
          cookie: auth.cookie,
        },
      }),
    ])

    const res = await client.query(qq, vars).toPromise()
    if (res?.error) {
      console.log('EXTENSION ERROR', res.error)
    }

    if (res?.extensions?.authorization?.accessToken) {
      auth.accessToken = res.extensions.authorization.accessToken
    }

    delete res.operation.context.fetch

    queryStore.data = res.data
    track(() => queryStore.data)

    const controller = new AbortController()
    cleanup(() => controller.abort())

    return queryStore.data
  })
}
```

The client is setup like this:

```
url: 'http://127.0.0.1:3001',
    requestPolicy: 'cache-first',
    fetchOptions: {
      credentials: 'include',
      headers: {
        cookie: serialize('jid', params.auth.cookie),
      },
    },
```


I am using the "Explorer" tab of urql dev tools.

My exchanges look like this:

```
exchanges: [
      devtoolsExchange,
      dedupExchange,
      cacheExchange({
        resolvers: {
          Query: {
            posts: relayPagination(),
          },
        },
    }),
    fetchExchange,
]
```

Everything works good except the graphcache part.
